database/src/desktop/query_desktop.cc

```diff
@@ -217,13 +217,14 @@ void QueryInternal::AddEventRegistration(
   database_->AddEventRegistration(query_spec_, listener_ptr,
                                   registration.get());
   Repo::scheduler().Schedule(NewCallback(
-      [](Repo::ThisRef ref, std::unique_ptr<EventRegistration> registration) {
+      [](Repo::ThisRef ref, EventRegistration* registration_released) {
         Repo::ThisRefLock lock(&ref);
+        std::unique_ptr<EventRegistration> registration(registration_released);
         if (lock.GetReference() != nullptr) {
           lock.GetReference()->AddEventCallback(std::move(registration));
         }
       },
-      database_->repo()->this_ref(), std::move(registration)));
+      database_->repo()->this_ref(), registration.release()));
```

Comment:
Two things:

First, I think we can move  `std::unique_ptr<EventRegistration> registration(registration_released);` one line above. Still have that gap of time but just slightly shorter. 

Second, I think the future fellows, including me, will feel confused about this block of code.
Better to add a comment like:
```
// Workaround to pass EventRegistration into scheduler callback since currently NewCallback  "copy", instead of "move",  every parameters.
// TODO(some bug number): Refactor NewCallback to support move operator.
```

Reply:
Take a look at the [new version](https://github.com/firebase/firebase-cpp-sdk/pull/1365/files/5355b5b871426987232fd410a4153495b3710f7a#diff-196c04498fc2dbafb32b5dd5f5076451a6cb3ee4a8e8bf40c1c348453a986947R220-R236) of this fix, it's much safer than the workaround above, and I think it's probably fine to keep as-is.

Reply:
I added a more complete comment about what is happening here, and added the TODO.